app/controllers/bulk_tagging/update_tags_controller.rb

```diff
@@ -0,0 +1,49 @@
+module BulkTagging
+  class UpdateTagsController < ApplicationController
+    def create
+      if valid_tag_update?
+        tag_migration.save!
+        redirect_to tag_migration_path(tag_migration)
+      else
+        redirect_to bulk_tagging_collection_path(params[:collection_content_id])
+      end
+    end
+
+  private
+
+    def tag_migration
+      @tag_migration ||= BulkTagging::BuildTagMigration.perform(
+        original_link_content_id: params[:collection_content_id],
+        taxon_content_ids: taxon_content_ids,
+        content_base_paths: content_base_paths
+      )
+    end
+
+    def taxon_content_ids
+      params[:taxons]
+    end
+
+    def content_base_paths
+      params[:content_base_paths]
+    end
+
+    # TODO: make this more readable, weird pattern
```

Comment:
Could this validation be moved into `BulkTagging::BuildTagMigration`? It looks like a form object anyway.